src/model/src/ModelTransformers.cpp

```diff
@@ -555,4 +555,73 @@ bool createModelWithNormalizedJointNumbering(const Model& model,
 }
 
 
+bool extractSubModel(const iDynTree::Model& fullModel, const iDynTree::Traversal& subModelTraversal,
+                     iDynTree::Model& outputSubModel)
+{
+    if (&subModelTraversal == NULL)
+    {
+        std::cerr << "[ERROR] subModelTraversal is null" << std::endl;
+        return false;
+    }
```

Comment:
We tipically do not check the validity of references as to pass an invalid reference you would need to dereference an invalid pointer, and that would raise an error before (I think).